File: prozesse_und_threads/loesung/03_execl.py
#!/usr/bin/env python3

import os

if __name__ == "__main__":
    print("Das ist der Beginn vom Prozess.")
    try:
        os.execl("/bin/ls1", "-la", "./")
    except FileNotFoundError:
        print("Kommando konnte nicht gefunden werden!")

    print("Zwischenausgabe zur Kontrolle.")
    os.execl("/bin/ls", "-la", "./ordern-existiert-nicht")
    print("Das ist der Rest vom Prozess.")

File: prozesse_und_threads/loesung/03_execl.cpp
#include <iostream>
#include <unistd.h>

using namespace std;

int main()
{
    cout << "Das ist der Beginn vom Prozess." << endl;
    execl("/bin/lsexistiertnicht", "-la", 0); // fehlerhafter Pfad
    cout << "Das ist der Rest vom Prozess." << endl;
    exit(0);
}